tangelo/algorithms/variational/tests/test_sa_vqe_solver.py

```diff
@@ -137,6 +138,31 @@ def test_simulate_h2(self):
         energy = vqe_solver_2.simulate()
         self.assertAlmostEqual(exact_energies[2], energy, delta=1.e-3)
 
+    def test_projected_simulate_h2(self):
+        """Run SA-VQE on H2 molecule, with UpCCGSD ansatz, JW qubit mapping, ref_states and exact simulator.
+        Followed by deflation of two states calculated to determine next excited state.
+        """
+
+        sym_check = Circuit([Gate("CNOT", target=q, control=4) for q in {0, 2}] + [Gate("CNOT", target=q, control=5) for q in [1, 3]])
+        sym_check += Circuit([Gate("MEASURE", 4), Gate("MEASURE", 5)])
```

Comment:
Just confirming: these measurement gates are mid-circuit measurements, and precede the deflation circuits, correct ? Through the simulate_options you are efficiently post-selection / projecting to the subspace with measures 00 for qubits 4 and 5 here ?